archivist/assetattachments.py

```diff
@@ -0,0 +1,165 @@
+    """AssetAttachmentsClient
+
+    Access to attachments entities using CRUD interface. This class is usually
+    accessed as an attribute of the Archivist class.
+
+    Args:
+        archivist (Archivist): :class:`Archivist` instance
+
+    """
+
+    def __init__(self, archivist: "type_helper.Archivist"):
+        self._archivist = archivist
+        self._public = archivist.public
+        self._subpath = f"{archivist.root}/{ASSETATTACHMENTS_SUBPATH}"
+        self._label = f"{self._subpath}/{ASSETATTACHMENTS_LABEL}"
+
+    def __str__(self) -> str:
+        if self._public:
+            return "AssetAttachmentsClient()"
+
+        return f"AssetAttachmentsClient({self._archivist.url})"
+
+    def _identity(self, identity: str, attachment_id: str) -> str:
+        """Return fully qualified identity
+        If public then expect a full url as argument
+
+        identity looks like:
+
+        [https://app.rkvst.io/archivist][/v2/]assets/xxxxxxx
+
+        OR
+
+        [https://app.rkvst.io/archivist][/v2/]assets/xxxxxxx/events/yyyyyy
+
+        """
+        uuid = attachment_id.split(SEP)[1]
+        if self._public:
+            url = urlparse(identity)
+            root = "/".join(url.path.split(SEP)[:2])
+            asset_id = "/".join(url.path.split(SEP)[2:])
+            new_url = url._replace(path=f"{root}{self._label}/{asset_id}/{uuid}")
+            return new_url.geturl()
+
+        return f"{self._label}/{identity}/{uuid}"
+
+    def __params(self, params: Optional[Dict]) -> Dict:
+        params = deepcopy(params) if params else {}
+        # pylint: disable=protected-access
+        return _deepmerge(self._archivist.fixtures.get(ATTACHMENTS_LABEL), params)
+
+    def download(
+        self,
+        identity: str,
+        attachment_id: str,
+        fd: BinaryIO,
+        *,
+        params: Optional[Dict] = None,
+    ) -> Response:
+        """Read attachment
+
+        Reads attachment into data sink (usually a file opened for write)..
+        Note that returns the response as the body will be consumed by the
+        fd iterator
+
+        Args:
+            identity (str): identity
+            attachment_id (str): blobs/aaaaaaaaaaaaa
+            fd (file): opened file descriptor or other file-type sink..
+            params (dict): e.g. {"allow_insecure": "true"} OR {"strict": "true" }
+
+        identity has one of the following 4 forms:
+
+            [https://app.rkvst.io/archivist/v2/] - if public
+            assets/xxxxxxxxxxxxxxxxxxxxxxxxxx
+
+            [https://app.rkvst.io/archivist/v2/] - if public
+            assets/xxxxxxxxxxxxxxxxxxxxxxxxxx/events/yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
+
+        Returns:
+            REST response
+
+        """
+
+        return self._archivist.get_file(
+            self._identity(identity, attachment_id), fd, params=self.__params(params)
+        )
+
+    def info(
+        self,
+        identity: str,
+        attachment_id: str,
+    ) -> Response:
```

Comment:
Return type should be `dict` as session.get returns `response.json()`

Reply:
fixed